be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/ProcessMessagesHandler.scala

```diff
@@ -49,6 +49,29 @@ object ProcessMessagesHandler extends AskPatternConstants {
     case value @ _ => value
   }
 
+  def rumorStateAnsHandler(messageRegistry: MessageRegistry)(implicit system: ActorSystem): Flow[GraphMessage, GraphMessage, NotUsed] = Flow[GraphMessage].map {
+    case msg @ Right(JsonRpcResponse(_, Some(resultObject), None, jsonId)) =>
+      resultObject.resultRumor match
+        case Some(rumorList) =>
+          val mergedMsg = rumorList.flatMap(rumor => rumor.messages)
+            .map((channel, msgList) => (channel, msgList.toSet))
+            .foldLeft(Map.empty[Channel, Set[Message]]) { case (acc, (key, values)) =>
+              acc.updated(key, acc.getOrElse(key, Set.empty[Message]) ++ values)
+            }
+          processMsgMap(mergedMsg, messageRegistry)
```

Comment:
maybe it's better to simplify it like that, it should lead to the same result 
val mergedMsg = rumorList
  .flatMap(_.messages)
  .groupBy(_._1)
  .mapValues(_.flatMap(_._2).toSet)

Reply:
I updated with a small difference because mapValues is deprecated in scala3.